### PR TITLE
COMPAT: fix f2py callbacks for objects with extra __call__ and __code__

### PR DESCRIPTION
**Pull request by mattip**
fixes issue #9383 where builtin functions have a ``__call__`` attribute in PyPy

### CODE REVIEW

**Comment by eric-wieser** [comment]
Not related to this patch, but I have no clue what the author was thinking when they decidedthat `\t` was the right thing to use here. Am I missing a reason not just to use either a literal tab, or spaces to match the rest of the code?

**Comment by eric-wieser** [comment]
Ok, you'll need to start over now that #9403 is merged - sorry

**Comment by mattip** [comment]
What's up with AppVeyor - why couldn't it merge when Travis could?

**Comment by eric-wieser** [comment]
Probably because `master` changed between the two runs starting

**Comment by mattip** [comment]
ok, but maybe next time merge/reject my pull request before making me redo it because of yours?

**Comment by eric-wieser** [comment]
Actually, I don't think that is the reason - this already had a failure cross next to it by the time mine was merged

**Comment by mattip** [comment]
Is there a f2py POC that should be CCed on this pull request?

File: numpy/f2py/cfuncs.py [hunk 1]
```diff
@@ -1045,17 +1045,23 @@
...
         }
+        else if (PyFortran_Check(fun) || PyFortran_Check1(fun)) {
+            tot = maxnofargs;
+            if (xa != NULL)
+                tot += PyTuple_Size((PyObject *)xa);
+            tmp_fun = fun;
+        }
         else if (PyObject_HasAttrString(fun,\"__call__\")) {
             tmp = PyObject_GetAttrString(fun,\"__call__\");
             if (PyObject_HasAttrString(tmp,\"im_func\"))
```
Comment (eric-wieser) [inline numpy/f2py/cfuncs.py]: This was renamed to `__func__` in python 3 - so is almost certainly wrong here, but I have no idea what it's even testing.

**Comment by pearu** [comment]
There seems to be some vagueness what create_cb_arglist should do. The basic rules of handling the arguments of callback functions are described in
    https://docs.scipy.org/doc/numpy-dev/f2py/python-usage.html#call-back-arguments
(see "Resolving arguments to call-back functions"). Let me know if you need more feedback.

**Comment by mattip** [comment]
PyPy3 now passes this test

**Pull request closed without merging**